packages/babel-core/src/transformation/file/generate.js

```diff
@@ -1,25 +1,36 @@
 // @flow
 
+import type { PluginPasses } from "../../config";
 import convertSourceMap, { type SourceMap } from "convert-source-map";
 import sourceMap from "source-map";
 import generate from "@babel/generator";
 
 import type File from "./file";
 
 export default function generateCode(
+  pluginPasses: PluginPasses,
   file: File,
 ): {
   outputCode: string,
   outputMap: SourceMap | null,
 } {
   const { opts, ast, shebang, code, inputMap } = file;
 
-  let gen = generate;
-  if (opts.generatorOpts && opts.generatorOpts.generator) {
-    gen = opts.generatorOpts.generator;
+  let result;
+  plugins_check: for (const plugins of pluginPasses) {
+    for (const plugin of plugins) {
+      const { generatorOverride } = plugin;
+      if (generatorOverride) {
+        result = generatorOverride(ast, opts.generatorOpts, code, generate);
+
+        if (result != null) break plugins_check;
```

Comment:
If two plugins provide two generators, the second one will be ignored?

Reply:
Yeah, I was thinking it gave some plugins the ability to toggle behavior more easily? Like

Reply:
Guess it ends up being essentially the same as https://github.com/webpack/tapable/blob/v0.2.8/lib/Tapable.js#L124 except my version allows `null`. Maybe it'd be better to only bail on specifically `undefined` though to be consistent with that.

Reply:
I'd prefer it to throw if two different parsers/generators don't return null, instead of using the first valid result.

Reply:
> Maybe it'd be better to only bail on specifically undefined though to be consistent with that.

How is null handled in other places in Babel options?

Reply:
> I'd prefer it to throw if two different parsers/generators don't return null, instead of using the first valid result.

Reply:
> How is null handled in other places in Babel options?

Reply:
Done. Changed to `undefined` and now throws an error of two different plugins return a result.